Updated Status in schema

thisistheplace opened the pull request:
Hi,

I've updated the Status model in the Schema to match the number type in the Fronius API docs, and removed it from the PvSystem model (since it's not a field in the Fronius docs).

Thanks,

Ben

drc38 commented:
Hi @thisistheplace 

Can you provide api references for these changes please. From the Api pdf it shows the following:

```
{ "pvSystemId": "04d81b82-7861-4e36-8e7f-41036ce711a4", "status": { "isOnline": true, "battMode": "1.0"  }, "data": { "logDateTime": "2019-06-18T14:01:57Z", "channels": [      { "channelName": "PowerFeedIn", "channelType": "Power", "unit": "W", "value": -496.01      },      { "channelName": "PowerLoad", "channelType": "Power", "unit": "W",
```

thisistheplace commented:
Hi @drc38,

I've been using the schema here as a reference:
[https://api.solarweb.com](https://api.solarweb.com/swqapi/index.html?utm_source=solar-web-system-mail&utm_medium=email&utm_campaign=SE__Solar-web-query-api-mails&utm_content=)

Which looks like it matches up with the current pdf docs on the Fronius site.

The changes I've made look like they match the data structure you've quoted?

I've removed the `status` field from the PvSystemMetaData since it isn't in the API schema, and changed the `battMode` type from a string to float (as per the `EnergyFlowStatus` schema).

Am I missing something?

Thanks!

thisistheplace commented:
Hi @drc38,

I've been using the `EnergyFlowStatus` schema here as a reference:
[https://api.solarweb.com](https://api.solarweb.com/swqapi/index.html?utm_source=solar-web-system-mail&utm_medium=email&utm_campaign=SE__Solar-web-query-api-mails&utm_content=)

Strange that the pdf states that the `status` field is a string, whereas the return type from the API Schema is actually a `number($double)`. 

There seems to be a discrepancy in the pdf docs (`status` field type) between the example response output on page 82 and page 85. I'm inclined to believe the Swagger docs rather than the pdf though.

What do you think?

drc38 commented:
> Hi @drc38,
> 
> I've been using the `EnergyFlowStatus` schema here as a reference: [https://api.solarweb.com](https://api.solarweb.com/swqapi/index.html?utm_source=solar-web-system-mail&utm_medium=email&utm_campaign=SE__Solar-web-query-api-mails&utm_content=)
> 
> Strange that the pdf example suggests that the `status.battMode` field is a string, whereas the return type from the API Schema is actually a `number($double)`. Although in page 81 it states that the type is a `number($double)`.
> 
> There seems to be a discrepancy in the pdf docs (`status.battMode` field type) between the example response output on page 82 and page 85. I'm inclined to believe the Swagger docs rather than the pdf though.
> 
> What do you think?

Thanks, could you update the PR so battMode is `(str | float)` just in case the api changed at some point and there's still firmware sending a string. Otherwise LGTM.

Thanks for your PRs

thisistheplace commented:
That's done. Thanks for writing the package!
